### PR TITLE
Tin backend summarize

### PR DESCRIPTION
**Pull request by tindo2003**

### CODE REVIEW

**Comment by nhanaa** [comment]
@tindo1234 Please merge main với lại reorganize cái service và route files hộ a cái.

File: backend/app/routes/folder.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from flask import Blueprint, request, jsonify
-from services.folder import FolderService
+from ..services.folder import FolderService
```
Comment (nhanaa) [inline backend/app/routes/folder.py]: Tại sao chuyển thành ..services.folder here?

File: backend/app/services/link_recommendation.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            my_dict["url"] = url_in_str
+            my_dict["website_vector_field"] = huggingface_embeddings.embed_query(my_str)
+            collection.insert_one(my_dict)
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@app.get("/test")
+def test_endpoint():
+    return {"message": "This is a test"}
```
Comment (nhanaa) [inline backend/app/services/link_recommendation.py]: File này có cần k? Thấy same với link_rec.py mà đk?

File: backend/app/routes/link_rec.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+takes in {"my_list" : [visitdatas]}
+outputs nothing
+"""
+
+
+@link_rec_bp.route("/insert_browsing_history", methods=["POST"])
+def insert_history():
+    data = request.get_json()
+    link_rec_service.insert_browsing_history(data=data["my_list"])
+    return jsonify({"message": "inserted successfully"})
```
Comment (nhanaa) [inline backend/app/routes/link_rec.py]: Em explain cái route này để làm gì đc k? Em có đang work với lại cái browsing history mà anh đã tạo k?

File: backend/app/services/link_recommendation.py (hunk 2 above)
Comment (tindo2003) [reply backend/app/services/link_recommendation.py]: Em deleted, thanks anh

File: backend/app/routes/link_rec.py (hunk 3 above)
Comment (tindo2003) [reply backend/app/routes/link_rec.py]: cái này em tạo chỉ để test cái link recommendation coi có work không :)

**Comment by tindo2003** [comment]
# PR Comment (in progress) 
     
## Chat Functionality
- **services/ai_services/rag.py**: Retrieves the correct documents, filtered by user_id and folder_id.
- **services/ai_services/agent.py**: An agent that decides which tool to use.
- **services/ai_services/chat.py**: Responsible for calling the agent in agent.py; invoked by /routes/chat.py.

## Link Recommendations upon Folder Creation (services/ai_services/link_rec.py)
- Looks for links in the `browsing_history` collection, filtered by IP address, similar to the user's folder title and description.

## Document Processing (services/ai_services/document.py)
**Function process_one_document**:

- Uses **summarize_website** in services/ai_services/summarization.py:
    - In summarize_website, divides website content into chunks and summarizes each chunk.
    - Uses LLM to generate a summary of the entire website based on the chunks' summaries. This method is necessary because website content can be very long, exceeding the LLM's token limit.
- Inserts original doc + summary and title, as suggested by the LLM, into the `documents` collection.
- Inserts the document's chunks, with contents appended with the entire document summary, into the `chunks` collection (inside services/ai_services/chunk.py). We do this to assist with the RAG process.

## Folder Summary (services/folder.py)
- Queries documents in the `documents` collection within the current folder, along with the user's folder description and title in a POST request, to generate a comprehensive summary (refer to routes/folder.py).

## Extra
- **services/ai_services/prompt.py**: Contains all kinds of prompts available for use.
- **services/ai_services/vectorstore_manager**: ensure correct vectorstore initialization and proper indexes, bridge between services and custom_vectorstore.py in utils.

**Pull request merged**